Add SOC2 section to security page (#801)
neondatabase/website: content/docs/security/security-overview.md

At Neon, security is our highest priority. We are committed to implementing best practices and earning the trust of our users.

## SOC2

> At Neon, we're proud to announce that we've successfully undergone a SOC2 Type 1 audit. SOC2 (Service Organization Control 2) Type 1 report is an attestation by an independent auditing entity, acknowledging that an organization has processes and systems in place that comply with the principles of the American Institute of Certified Public Accountants (AICPA).

Reviewer: ```suggestion
Neon has successfully completed a SOC2 Type 1 audit. SOC2 (Service Organization Control 2) Type 1 report is an attestation by an independent auditing entity, acknowledging that an organization has processes and systems in place that comply with the principles of the American Institute of Certified Public Accountants (AICPA).
```

This audit report focuses on five trust service criteria: security, availability, processing integrity, confidentiality, and privacy. Our successful completion of this audit signifies that our design of systems and procedures, as of a specific date, meet these stringent criteria.

> Please note that while the term "SOC2 Certified" is commonly used, it's not accurate and we refrain from using it. The AICPA explicitly advises against using this term. A SOC2 report is an attestation of the effectiveness of controls, not a certification.

Reviewer: ```suggestion
Please note that while the term "SOC2 Certified" is commonly used, it is not accurate and we refrain from using it. The AICPA explicitly advises against using this term. A SOC2 report is an attestation of the effectiveness of controls, not a certification.
```

> Our SOC2 Type 1 report underscores our ongoing commitment to safeguarding our customers' data and maintaining a high level of security. We will continue to invest in regular audits and adhere to industry best practices to ensure the security, integrity, and privacy of the data you entrust us with.

Reviewer: Thanks for updating, @zoete . A few minor suggestions above.
